### PR TITLE
LPS-124711 - Update clay-* dependencies to 2.22.0

### PR DESCRIPTION
**Pull request by bryceosterhaus**
This updates clay v2 dependencies to 2.22.0

### CODE REVIEW

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Review by jbalsas — CHANGES_REQUESTED** [review]
Hey @bryceosterhaus, when I saw this I wondered if we could simply stop bundling our `clay-` packages. I think it's about time we do that :)

I checked and most of those usages in `package.json` are stale. Those dependencies were at some point migrated towards `@clayui/*` but the old ones were never removed. 

Others are soon to go away like `flags-taglib` and maybe we could have @matuzalemsteles and @diegonvs take a look at those in `dynamic-data-mapping`. 

All this to say, before pushing forward, could you please check and remove the unnecessary dependencies in the `package.json` of the modules so we have a clearer picture of what's still really necessary?

Thanks!! :)

File: modules/yarn.lock [hunk 1]
```diff
@@ -4216,71 +4216,82 @@ clay-icon@2.21.4, clay-icon@^2.21.4:
     metal-state "^2.16.0"
     metal-web-component "^2.16.0"
 
-clay-label@2.21.4, clay-label@^2.21.4:
-  version "2.21.4"
-  resolved "https://registry.yarnpkg.com/clay-label/-/clay-label-2.21.4.tgz#5495aff28caa9dcff3d63bafc611b5ec60e6e0a0"
-  integrity sha512-DCgXL4pQTfqlntpNacjw0njGPo3jHLEn5+mCJHfYgmgZJ/SNBDtOD7syFmuo60vY/KfXSABOw6NNnUKU1UbZBA==
+clay-icon@2.22.0, clay-icon@^2.21.4, clay-icon@^2.22.0:
```
Comment (jbalsas) [inline modules/yarn.lock:4219]: Wonder where's this fixed dependency coming from... my bet, commerce is sneaking a lot of packages still outside of SF that weren't updated maybe?

Comment (jbalsas) [reply modules/yarn.lock:4219]: There are other packages also duplicated. Diff in addition/deletion in `yarn.lock` usually gives a hint about this ;)

**Comment by matuzalemsteles** [comment]
@jbalsas we are slowly removing v2 packages from DDM as we migrate to React and using most of the Clay components https://github.com/liferay/liferay-portal/commit/a93b1087a922eeddad86049377587322978c7613.

One thing I remember that separated Clay v2 from v3 more were some changes related to CSS, so maybe we still have to cut versions of v2 when we have to change things related to CSS?

File: modules/yarn.lock (hunk 1 above)
Comment (bryceosterhaus) [reply modules/yarn.lock:4219]: > commerce is sneaking a lot of packages still outside of SF that weren't updated maybe

Yeah you are right, I forgot they have to be manually updated due to SF suppression. I'll take a look!

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
@jbalsas 

- Updated commerce dependencies
- Removed un-used clay-* dependencies from modules consuming clay
- Removed un-used clay-* dependencies from `frontend-taglib-clay` that weren't being used in DXP at all.
 
Wasn't super sure about the last one, but thought it would help us lock in which packages are still used and an slowly get rid of them.

**Comment by bryceosterhaus** [comment]
Also, is the `frontend-taglib-chart` module used? I couldn't find uses of it and thought if we removed that we could also rid ourselves of charts 2.x

**Comment by bryceosterhaus** [comment]
cc @john-co this kernal version error seems unrelated, can you check this out?

**Comment by john-co** [comment]
Hi @bryceosterhaus , yes it's unrelated. Looks like the same as what was discussed in slack: https://liferay.slack.com/archives/CL84ZPHAT/p1607614841469600

**Comment by john-co** [comment]
The issue was fixed in https://github.com/brianchandotcom/liferay-portal/pull/96806, probably better to just manually forward.

**Comment by antonio-ortega** [comment]
Hi guys,

This issue will be sooner rather than later backport it to 7.2.x where we need all those clay-* dependencies. I'd like to suggest you to use two separate LPSs for the two tasks you are carrying out:

- Update clay-* to 2.22.0
- Remove unneeded clay-* dependencies in Master

That way the backport of the first point to old branches will be much easier and can be done sooner.

Thanks in advance.

Regards.

File: modules/apps/commerce/commerce-cart-taglib/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
 	"dependencies": {
-		"clay-icon": "2.21.4",
+		"clay-icon": "2.20.0",
```
Comment (jbalsas) [inline modules/apps/commerce/commerce-cart-taglib/package.json:3]: This should be `2.22.0` not `2.20.0` same as others

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
Hey @antonio-ortega, I've amended the commits to avoid removing `clay-` packages for now. I've also tried to split the changes in different commits so it's easier to backport but I'm not positive it will be since commerce is not present in `7.2` and changes will likely need to be manually recreated.

**Comment by jbalsas** [comment]
Errors seem unrelated... going to give this another try

/cc @john-co, @PablitoBonito, @manoelcyreno

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
Well... looks like CI is not on our side on this one... will forward manually

**Comment by jbalsas** [comment]
ci:stop

**Comment by jbalsas** [comment]
And it now conflicts with brian's upstream... this is honestly becoming very frustrating :(

**Comment by jbalsas** [comment]
Rebased and pushed manually at https://github.com/brianchandotcom/liferay-portal/pull/96862

**Comment by bryceosterhaus** [comment]
> And it now conflicts with brian's upstream

@jbalsas, would it be worth syncing this repo with Bchans upsteam rather than liferay/liferay-portal?

**Comment by jbalsas** [comment]
> @jbalsas, would it be worth syncing this repo with Bchans upsteam rather than liferay/liferay-portal?

Well, that makes for a longer conversation 😂 

The thing is `liferay/liferay-portal` is the canonical and "stable" repo. Syncing with bchan's exposes us to more instability since code merged there can be in all sorts of unstable fashion until it makes its way to upstream.

**Comment by wincent** [comment]
I made https://github.com/liferay/clay/issues/3849 as a place to discuss one particular obvious (but radical) solution to this problem.

**Pull request closed without merging**